File: algorithms/python/14-longest-common-prefix/solution1.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        
        shortest = min(strs, key=len)
        for i, ch in enumerate(shortest):
            for str in strs:
                if str[i] != ch:
                    return shortest[:i]
        
        return shortest

File: algorithms/14-longest-common-prefix/go/solution1.go
func longestCommonPrefix(strs []string) string {
    ret := ""

	for i in range(len(strs[0])) {
		for j in range(1, len(strs)) {
			if i >= len(strs[j]) || strs[j][i] != strs[0][i] {
				return ret
			}
		}
		ret += string(strs[0][i])
	}

	return ret
}